sound/soc/sof/intel/byt.c

```diff
@@ -636,6 +636,14 @@ const struct snd_sof_dsp_ops sof_tng_ops = {
 };
 EXPORT_SYMBOL(sof_tng_ops);
 
+const struct sof_intel_dsp_desc tng_chip_info = {
+	/* Tng */
+	.cores_num = 1,
+	.cores_mask = 1,
+	.ops = &sof_tng_ops,
+};
```

Comment:
Ah this is interesting. There are multiple concepts that are not aligned here.
1. sof_intel_dsp_desc is mostly for SKL+ devices
2. acpi devices assign chip_info to the ops
3. but here in the TNG case we use the sof_intel_dsp_desc because it's PCI.

I also wonder if we could improve the Baytrail/CherryTrail/HSW/Broadwell IPC by using a descriptor pointing to the actual registers instead of hard-coding them, .e.g. generalizing sof_intel_dsp_desc to also work with legacy platforms.

Really feels like we've done most of the work but there are still improvements we could do here.

Reply:
yes, I also agree with you. First I tried to use sof_intel_dsp_desc in ACPI, but I found sof_intel_dsp_desc is define in hda.h. Now for sof_intel_dsp_desc has been moved to shim.h, I will use sof_intel_dsp_desc to ACPI.